lib/ExecutionEngine/ExecutionEngine2.cpp

```diff
@@ -45,6 +45,9 @@ void ExecutionEngine2::setBackendName(llvm::StringRef backend) {
   if (backend != "") {
```

Comment:
what's the use case when backend passed here is actually an empty string?

Reply:
You ask a good question.  It's currently called in  ~ExecutionEngine2, but the module without a HM doesn't make much sense. A follow up Diff to refactor and simplify seems like a good idea.

Reply:
Made #3280 to track it. I'll refactor when I have fewer branches to rebase :)